test/form/samples/class-name-not-changed/_expected/amd.js

```diff
@@ -0,0 +1,17 @@
+define(['exports'], function (exports) { 'use strict';
+
+	class MyClass {
+		constructor() { }
+	}
+
+	let MyClass$1 = class MyClass {
+		constructor() { }
+	}
```

Comment:
Actually I think this is incorrect if we don't have automatic semicolon insertion here.

For example:

```js
class Q {
} p();
```

is valid JavaScript, while:

```js
let Q$1 = class Q {
} p()
```

is not!

Reply:
Good catch. Should I just unconditionally add a semicolon (or maybe a new line)? From what I could tell, it's not possible to check whether the next token is preceded by a new line without doing additional parsing manually.

Reply:
See https://github.com/rollup/rollup/blob/master/src/ast/nodes/ExportDefaultDeclaration.ts#L112 for a rough example of how this can work. I think the crux is just:

```js
if (code.original[this.end] !== ';') {
  code.prependRight(this.end, ';');
}
```